store/persistent/storethehash/storethehash_test.go

```diff
@@ -40,3 +44,60 @@ func TestRemoveMany(t *testing.T) {
 	}
 	persistent.RemoveManyTest(t, s)
 }
+
+func TestPeriodicFlush(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.SkipNow()
+	}
+	// Init storage
+	tmpDir, err := ioutil.TempDir("", "sth")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	s, err := storethehash.New(tmpDir)
+	if err != nil {
+		t.Fatal(err)
+	}
+	p, err := peer.Decode("12D3KooWKRyzVWW6ChFjQjK4miCty85Niy48tpPV95XdKu1BcvMA")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Put some data in the first storage.
+	cids, err := utils.RandomCids(151)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	entry := store.MakeIndexEntry(p, 0, cids[0].Bytes())
+	for _, c := range cids[1:] {
+		_, err = s.Put(c, entry)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// Sleep for 2 sync Intervals to ensure that data is flushed
+	time.Sleep(2 * storethehash.DefaultSyncInterval)
+	s.Close()
```

Comment:
I think that calling `Close` should cause a flush immediately and that there should not be any need to wait before closing.  If there is a need to wait before closing then that should be built into the `Close`

Reply:
This `s.Close()` is not necessary here. I forgot to remove it from the test after doing some sanity-checks. `s.Close()` calls storethehash's `Close` method which is the one [responsible for immediately flushing](https://github.com/ipld/go-storethehash/blob/1ea50cc878b8f7f10f79207a8060d950edb8a0e8/store/store.go#L106). Already fixed it. Thanks!